index.js

```diff
@@ -96,23 +96,32 @@ module.exports = {
     
     self.getNewOptionValue = function(req, moduleOptions, sliced, val) {
       if (val && (typeof(val) === 'object')) {
+        var comparator;
+        if (typeof(val.comparator) === 'string') {
+          comparator = function(value, other) {
+            return value[val.comparator] === other[val.comparator];
+          }
+        } else if (typeof(val.comparator) === 'function') {
+          comparator = val.comparator;
+        } else {
+          comparator = _.isEqual;
+        }
+
         if (val.$append) {
           array = _.get(moduleOptions, sliced) || [];
           return array.concat(val.$append);
         } else if (val.$prepend) {
           array = _.get(moduleOptions, sliced) || [];
           return val.$prepend.concat(array);
         } else if (val.$appendUnique) {
-          array = _.get(moduleOptions, sliced) || [];
-          added = _.differenceWith(val.$appendUnique, array, _.isEqual);
-          return array.concat(added);
+          array = (_.get(moduleOptions, sliced) || []).concat(val.$appendUnique);
+          return _.reverse(_.uniqWith(_.reverse(array), comparator));
         } else if (val.$prependUnique) {
-          array = _.get(moduleOptions, sliced) || [];
-          added = _.differenceWith(val.$prependUnique, array, _.isEqual);
-          return added.concat(array || []);
+          array =  val.$prependUnique.concat(_.get(moduleOptions, sliced) || []);
+          return _.uniqWith(array, comparator);
```

Comment:
@boutell  This is a "breaking" change, assuming the value I want to `prepend` will be unique on the head of the array. Assuming the user want to force move a value in the array.

Reply:
Forcing the move seems incorrect to me, also this code would throw out all duplicates, not just duplicates that are relevant to the list in prependUnique or appendUnique.